docs/spec/slashing/tombstone.md

```diff
@@ -0,0 +1,74 @@
+# Staking Tombstone
+
+## Abstract
+
+In the current implementation of the `slashing` module, when the consensus engine informs the state machine of a validator's consensus fault, the validator is partially slashed, and put into a "jail period", a period of time in which they are not allowed to rejoin the validator set.  However, because of the nature of consensus faults and ABCI, there can be a delay between an infraction occuring, and evidence of the infraction reaching the state machine (this is one of the primary reasons for the existence of the unbonding period).
+
+In the current system design, once a validator is put in the jail for a consensus fault, after the `JailPeriod` they are allowed to send a transaction to `unjail` themselves, and thus rejoin the validator set.
+
+One of the "design desires" of `slashing` module is that if multiple infractions occur before evidence is executed (and a validator is put in jail), they should only be punished for single worst infraction, but not cumulatively.  For example, if the sequence of events is:
+1. Validator A commits Infraction 1 (worth 30% slash)
+2. Validator A commits Infraction 2 (worth 40% slash)
+3. Validator A commits Infraction 3 (worth 35% slash)
+4. Evidence for Infraction 1 reaches state machine (and validator is put in jail)
+5. Evidence for Infraction 2 reaches state machine
+6. Evidence for Infraction 3 reaches state machine
+   
+Only Infraction 2 should have it's slash take effect, as it is the highest.  This is done, so that in the case of the compromise of a validator's consensus key, they will only be punished once, even if the hacker double-signs many blocks.  Because, the unjailing has to be done with the validator's account key, they have a chance to resecure their consensus key, and then signal that they are ready using their account key.  We call this period during which we track only the max infraction, the "slashing period".
+
+> Note:  Currently, from what I can tell in the code, while the stake actively delegated to a validator follows the Max slash rule, but stake that is unbonding or redelegating away from a validator gets slashed cumulatively for consecutive faults.  I think this may be a bug that needs to be fixed independently of whether the tombstone is accepted or not.
```

Comment:
I don't think this is true; the slashing periods caps slashing per-validator before any redelegations or unbonding delegations are slashed.

It is true that redelegating stake can be liable for two validators' infractions at once, if that's what you mean.

(I could be wrong though - can you point to what part of the code you read?)

Reply:
Oh never, mind.  I realized that the `revisedFraction` returned by the `capBySlashingPeriod` function in `x/slashing` isn't the new MaxSlash amount, but rather the difference b/w MaxSlash and CurrentSlash.  So never mind, it works correctly right now.